### ISSUE TITLE
Clarification on the Implementation of Influence Score

### ISSUE THREAD

**Issue by 311dada** [cxcscmu/MATES#5]
Congratulations on your excellent work. While reading your code, I found myself a bit confused about the implementation of the influence score. Specifically, equation (12) suggests that the influence score of a sample is positively correlated with the **change in the validation set loss** after one-step training on that sample. However, in the open-source implementation, the influence score seems to be based **on the validation set loss itself, rather than the change in loss**. Relevant codes are
* [oracle influence collection code](https://github.com/cxcscmu/MATES/blob/5d2860c4c26291c26375a0689e33e133b6b4978b/src/select_data/probe_oracle_data_influence.py#L224) 
* [influence model training code ](https://github.com/cxcscmu/MATES/blob/5d2860c4c26291c26375a0689e33e133b6b4978b/src/select_data/train_data_influence_model.py#L54)

Am I misunderstanding something, or is there an issue with the implementation? Could you provide some clarification?

Thank you!

**Comment by yuzc19** [comment]
Hi @311dada,

This is a great question! In the paper, we formulate the oracle influence as the change in the validation loss before and after one-step training to match the original influence function. But empirically, the validation loss before one-step training is a **constant** for all probing data, so we remove this term for simplicity. In the [implementation code of selection](https://github.com/cxcscmu/MATES/blob/main/src/select_data/select_data.py), we select the data that can achieve the **lowest** predicted validation loss, so there is no actual issue with the code.

**Comment by 311dada** [comment]
Thank you for your reply. However, I still have two questions:

* In one-step training, should we restore the initial model each time to ensure that the validation loss remains constant?
* If we simplify the influence score to the validation loss, as you mentioned, a smaller score indicates that the sample is more important. However, in the [code](https://github.com/cxcscmu/MATES/blob/5d2860c4c26291c26375a0689e33e133b6b4978b/src/select_data/select_data.py#L38), the top-k is selected. Could this be problematic?

**Comment by yuzc19** [comment]
For 1: We do resume the model state each time [here](https://github.com/cxcscmu/MATES/blob/5d2860c4c26291c26375a0689e33e133b6b4978b/src/select_data/probe_oracle_data_influence.py#L182).
For 2: We select the data with the lowest metric scores [here](https://github.com/cxcscmu/MATES/blob/5d2860c4c26291c26375a0689e33e133b6b4978b/src/select_data/select_data.py#L40C5-L40C69).

**Comment by 311dada** [comment]
My fault! Thanks for your patience!
